### PR TITLE
[SS] Add debug logging in uffd

### PR DESCRIPTION
**Pull request by maggie-lou**
**Related issues**: buildbuddy-io/buildbuddy-internal#3092

### CODE REVIEW

**Comment by maggie-lou** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #5884** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/5884?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/5884?utm_source=stack-comment).

File: enterprise/server/remote_execution/uffd/uffd.go [hunk 1]
```diff
@@ -297,14 +334,37 @@ func (h *Handler) handle(ctx context.Context, memoryStore *copy_on_write.COWStor
...
+			InvalidBytesAtChunkStart: int64(invalidBytesAtChunkStart),
+			InvalidBytesAtChunkEnd:   invalidBytesAtChunkEnd,
+		}
+		if alreadyMappedRanges, alreadyMapped := h.mappedPageFaults[int64(chunkStartOffset)]; alreadyMapped {
+			alreadyMappedStr := ""
+			for _, alreadyMappedRange := range alreadyMappedRanges {
+				alreadyMappedStr += alreadyMappedRange.String()
+			}
+			alreadyMappedStr += debugData.String()
+			log.Warningf("Attempted to map chunk start offset 0x%x multiple times: %s", chunkStartOffset, alreadyMappedStr)
```
Comment (bduffany) [inline enterprise/server/remote_execution/uffd/uffd.go:356]:
```suggestion
			log.CtxWarningf(ctx, "Attempted to map chunk start offset 0x%x multiple times: %s", chunkStartOffset, alreadyMappedStr)
```

File: enterprise/server/remote_execution/uffd/uffd.go [hunk 2]
```diff
@@ -297,14 +334,37 @@ func (h *Handler) handle(ctx context.Context, memoryStore *copy_on_write.COWStor
...
+		}
+		h.mappedPageFaults[int64(chunkStartOffset)] = append(h.mappedPageFaults[int64(chunkStartOffset)], debugData)
+
 		_, err = resolvePageFault(uffd, uint64(destAddr), uint64(hostAddr), uint64(copySize))
 		if err != nil {
+			log.Warningf("Failed to resolve page fault %s due to err %s", debugData.String(), err)
```
Comment (bduffany) [inline enterprise/server/remote_execution/uffd/uffd.go:362]:
```suggestion
			log.CtxWarningf(ctx, "Failed to resolve page fault %s due to err %s", debugData.String(), err)
```

**Review by bduffany — APPROVED** [review]
nice!

**Pull request merged**